extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -29,118 +27,118 @@ const metadataProvider = classes.MetadataProvider;
-      // TODO -> We'll need to merge auth later.
-      qidoDicomWebClient = dicomWebConfig.staticWado
-        ? new StaticWadoClient(qidoConfig)
-        : new api.DICOMwebClient(qidoConfig);
-
-      wadoDicomWebClient = dicomWebConfig.staticWado
-        ? new StaticWadoClient(wadoConfig)
-        : new api.DICOMwebClient(wadoConfig);
+      clientManagerObj = new clientManager({
+        params,
+        query,
+        dicomWebConfig,
+        userAuthenticationService,
+      });
+      implementation.reject = clientManagerObj.reject();
     },
     query: {
       studies: {
         mapParams: mapParams.bind(),
         search: async function (origParams) {
-          qidoDicomWebClient.headers = getAuthrorizationHeader();
-          const { studyInstanceUid, seriesInstanceUid, ...mappedParams } =
-            mapParams(origParams, {
-              supportsFuzzyMatching: dicomWebConfig.supportsFuzzyMatching,
-              supportsWildcard: dicomWebConfig.supportsWildcard,
-            }) || {};
-
-          const results = await qidoSearch(qidoDicomWebClient, undefined, undefined, mappedParams);
-
+          clientManagerObj.setQidoHeaders();
+          let results = [];
+          const studyInstanceUIDs = {};
+          // concatenate series metadata from all servers
+          const clients = clientManagerObj.getClients();
+          for (let i = 0; i < clients.length; i++) {
+            const { studyInstanceUid, seriesInstanceUid, ...mappedParams } =
+              mapParams(origParams, {
+                supportsFuzzyMatching: clients[i].supportsFuzzyMatching,
+                supportsWildcard: clients[i].supportsWildcard,
+              }) || {};
+            let clientResults;
+            try {
+              clientResults = await qidoSearch(
+                clients[i].qidoDicomWebClient,
+                undefined,
+                undefined,
+                mappedParams
+              );
+            } catch {
+              clientResults = [];
+            }
+            for (let j = 0; j < clientResults.length; j++) {
+              const studyInstanceUID = clientResults[j]['0020000D'].Value[0];
+              if (!(studyInstanceUID in studyInstanceUIDs)) {
+                studyInstanceUIDs[studyInstanceUID] = clientResults[j];
+                results.push(clientResults[j]);
+              } else {
+                // updates number of series in study
+                if (
+                  studyInstanceUIDs[studyInstanceUID]['00201206'] &&
+                  clientResults[j]['00201206']
+                ) {
+                  studyInstanceUIDs[studyInstanceUID]['00201206'].Value[0] =
+                    studyInstanceUIDs[studyInstanceUID]['00201206'].Value[0] +
+                    clientResults[j]['00201206'].Value[0];
+                }
+
+                // updates number of instances in study
+                if (
+                  studyInstanceUIDs[studyInstanceUID]['00201208'] &&
```

Comment:
Please add a constant for `00201208` and ANY other DICOM tags that are in this file.